spec/rails_integration_spec.rb

```diff
@@ -106,14 +106,12 @@ def prepare_params(params)
   end
 
   describe "nested_array" do
-    it "responds with a 400 when the nested array is not supplied properly" do
-      params = {
-        'filter' => 'state'
-      }
+    it "responds with a 200 when the nested array is provided as nil" do
+      expect { get :nested_array, **prepare_params({ filter: { state: nil } }) }.not_to raise_error
+    end
 
-      expect { get :nested_array, **prepare_params(params) }.to raise_error do |error|
-        expect(error).to be_a(RailsParam::InvalidParameterError)
-      end
+    it "responds with a 200 when the nested array is processed as nil" do
+      expect { get :nested_array, **prepare_params({ filter: "state" }) }.not_to raise_error
```

Comment:
This results in `ParamEvaluator` being called with the following `params`:

```
#<ActionController::Parameters {"filter"=>"state", "controller"=>"fake", "action"=>"nested_array"} permitted: false>
{"state"=>nil}
```

And fails.